test/index.js

```diff
@@ -291,6 +297,96 @@ describe('instantiation', function() {
   });
 });
 
+describe('snapshot_() method', function() {
+  const bytesData = Buffer.from('AQI=', 'base64');
+  let firestore;
+
+  beforeEach(function() {
+    firestore = createInstance();
+  });
+
+  it('handles ProtobufJS', function() {
+    let doc = firestore.snapshot_(
+      document('doc', {
+        foo: {valueType: 'bytesValue', bytesValue: bytesData},
+      }),
+      {seconds: 5, nanos: 6}
+    );
+
+    assert.equal(true, doc.exists);
+    assert.deepEqual({foo: bytesData}, doc.data());
+    assert.equal('1970-01-01T00:00:01.000000002Z', doc.createTime);
+    assert.equal('1970-01-01T00:00:03.000000004Z', doc.updateTime);
+    assert.equal('1970-01-01T00:00:05.000000006Z', doc.readTime);
+  });
+
+  it('handles API JSON', function() {
```

Comment:
Maybe add a comment something like:

// Google Cloud Functions must be able to call snapshot_() with API JSON data.